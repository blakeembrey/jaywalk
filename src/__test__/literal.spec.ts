import { validationFunction } from '../'

describe('Literal', () => {
  it('should validate literals', () => {
    const validate = validationFunction({
      '@type': 'Literal',
      value: 'test'
    })

    const input = 'test'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-literals', () => {
    const validate = validationFunction({
      '@type': 'Literal',
      value: 'test'
    })

    expect.assertions(1)

    return validate(10).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
