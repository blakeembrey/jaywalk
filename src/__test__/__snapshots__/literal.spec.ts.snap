// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Literal should reject non-literals 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a literal",
    "schemaPath": Array [
      "@type",
    ],
  },
]
`;
